src/components/Search/SearchLayout.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Container,
+  createStyles,
+  Group,
+  Stack,
+  Text,
+  ThemeIcon,
+  Tooltip,
+  UnstyledButton,
+} from '@mantine/core';
+import { createContext, Dispatch, SetStateAction, useContext, useMemo, useState } from 'react';
+import { AppLayout } from '~/components/AppLayout/AppLayout';
+import { IconChevronsLeft } from '@tabler/icons-react';
+import { routing } from '~/components/Search/useSearchState';
+import { instantMeiliSearch } from '@meilisearch/instant-meilisearch';
+import { env } from '~/env/client.mjs';
+import { SearchIndex } from '~/components/Search/parsers/base';
+import { InstantSearch } from 'react-instantsearch';
+import { CustomSearchBox } from '~/components/Search/CustomSearchComponents';
+import { RenderSearchComponentProps } from '~/components/AppLayout/AppHeader';
+
+const SIDEBAR_SIZE = 377;
+
+const searchClient = instantMeiliSearch(
+  env.NEXT_PUBLIC_SEARCH_HOST as string,
+  env.NEXT_PUBLIC_SEARCH_CLIENT_KEY,
+  { primaryKey: 'id', keepZeroFacets: true }
+);
+
+// #region [ImageGuardContext]
+type SearchLayoutState = {
+  sidebarOpen: boolean;
+  setSidebarOpen: Dispatch<SetStateAction<boolean>>;
+};
+
+const SearchLayoutCtx = createContext<SearchLayoutState>({} as any);
+export const useSearchLayoutCtx = () => {
+  const context = useContext(SearchLayoutCtx);
+  if (!context) throw new Error('useSearchLayoutIdx can only be used inside SearchLayoutCtx');
+  return context;
+};
```

Comment:
Usually the convention for context hooks is to call it
```suggestion
export const useSearchLayout = () => {
  const context = useContext(SearchLayoutCtx);
  if (!context) throw new Error('useSearchLayout can only be used inside SearchLayoutCtx');
  return context;
};
```